subgraph/src/fleek-nfa.ts

```diff
@@ -257,6 +254,91 @@ export function handleMetadataUpdateWithIntValue(
   }
 }
 
+export function handleInitialized(event: InitializedEvent): void {
+    // This is the contract creation transaction.
+    log.warning('This is the contract creation transaction.', []);
+    if (event.receipt) {
+      let receipt = event.receipt as ethereum.TransactionReceipt;
+      log.warning('Contract address is: {}', [
+        receipt.contractAddress.toHexString(),
+      ]);
+ 
+      // add owner
+      let owner = new Owner(event.transaction.from);
+      owner.collection = true;
+      owner.save();
+    }
+}
+
+export function handleCollectionRoleChanged(event: CollectionRoleChangedEvent): void {
+  let toAddress = event.params.toAddress;
+  let byAddress = event.params.byAddress;
+  let role = event.params.role;
+  let status = event.params.status;
+  
+  if (role == 0) {
+    // Owner role
+    if (status) {
+      // granted
+      let owner = Owner.load(toAddress);
+      if (!owner) {
+        owner = new Owner(toAddress);
+      }
+      owner.collection = true;
+      owner.save();
+    } else {
+      // revoked
+      let owner = Owner.load(toAddress);
+      if (!owner) {
+        log.error('Owner entity not found. Role: {}, byAddress: {}, toAddress: {}', [role.toString(), byAddress.toHexString(), toAddress.toHexString()]);
+        return;
+      }
+      owner.collection = false;
+      owner.save();
+    }
+  } else {
+    log.error('Role not supported. Role: {}, byAddress: {}, toAddress: {}', [role.toString(), byAddress.toHexString(), toAddress.toHexString()]);
+  }
+}
+
+export function handleTokenRoleChanged(event: TokenRoleChangedEvent): void {
+  let tokenId = event.params.tokenId;
+  let toAddress = event.params.toAddress;
+  let byAddress = event.params.byAddress;
+  let role = event.params.role;
+  let status = event.params.status;
+  
+  // load token
+  let token = Token.load(Bytes.fromByteArray(Bytes.fromBigInt(tokenId)));
+  if (!token) {
+    log.error('Token not found. TokenId: {}', [tokenId.toString()]);
+    return;
+  } 
+
+  if (role == 0) {
+    // Controller role
+    // get the list of controllers.
+    let token_controllers = token.controllers;
```

Comment:
```suggestion
    const tokenControllers = token.controllers;
```

Reply:
we assign to this variable later in the if condition though